blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java

```diff
@@ -72,132 +70,106 @@ public BluePipeline create(Reachable parent) throws IOException {
         Preconditions.checkNotNull(parent, "Parent passed is null");
         Preconditions.checkNotNull(getName(), "Name provided was null");
 
-        String apiUrl = null;
-        String orgName = getName(); //default
-        String credentialId = null;
-        List<String> repos = new ArrayList<>();
+        final List<String> repos = getRepositories();
+        final String repositoryToAdd = repos.size() == 1 ? repos.get(0) : null;
+        final boolean isUpdatingSingleRepo = repositoryToAdd != null;
 
-        if (scmConfig != null) {
-            apiUrl = StringUtils.defaultIfBlank(scmConfig.getUri(), GithubScm.DEFAULT_API_URI);
-            if (scmConfig.getConfig().get("orgName") instanceof String) {
-                orgName = (String) scmConfig.getConfig().get("orgName");
-            }
-            credentialId = scmConfig.getCredentialId();
-            if (scmConfig != null && scmConfig.getConfig().get("repos") instanceof List) {
-                for (String r : (List<String>) scmConfig.getConfig().get("repos")) {
-                    repos.add(r);
+        Item item = Jenkins.getInstance().getItemByFullName(getName());
```

Comment:
@i386 We were avoiding creation of item till credentialId was valid, I guess its not too bad as we are deleting created item on error anyways, but add extra cleanup on bad request. Changes looks fine. I would say, there have been regressions in this part of code and most testing have been manual so please check with @cliffmeyers and @kzantow on testing involved before we know its all good. Or maybe keep this PR focused on unit test and this refactoring in separate PR? Up to you.